### PR TITLE
Add support for generating a timestamp

### LINKED ISSUE #187: Support generating a timestamp
It may be useful to be able insert a timestamp in the generated website.

e.g.,

`{{timestamp}}`
-->
><sub>[Generated by [MarkBind]() at 2018.03.30 22.50.34 UTC]</sub>

### PR DESCRIPTION
**Pull request by Chng-Zhi-Xuan**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**
• [X] Enhancement to an existing feature

Resolves #187 

**What is the rationale for this request?**
User would like an additional pre-defined variable within MarkBind to insert timestamps

**What changes did you make? (Give an overview)**
- Add a new Util function `initPageTimestamp(page)` that overrides value with the 'timestamp' key within `this.userDefinedVariablesMap[base]` with the current time.
- Function is called in `generatePages` and `regenerateAffectedPages`
- Update UserGuide [here](https://deploy-preview-222--markbind.netlify.com/userguide/contentauthoring#use-markbind-variables)
**Provide some examples**

Used in the same way as glyphicons | 
--- | 
feature-timestamp |

Known limitations |
--- |
Clicking the `MarkBind` link when the timestamp is placed in a component header will not work |
Color of link and panel type="primary" blends together as shown above |

**Is there anything you'd like reviewers to focus on?**
- code variable naming / style

**Testing instructions:**
1. Insert {{timestamp}} anywhere in the source,
1. Check it renders correctly
1. Click on the hyperlink within the generated timestamp and check it works.

### CODE REVIEW

File: docs/userGuide/contentAuthoring.md [hunk 1]
```diff
@@ -113,6 +113,34 @@ In addition, it supports:
 
 * More media blocks, embedding services and additional options can be found in [Markdown-it-block-embed docs](https://github.com/rotorz/markdown-it-block-embed)
 
+### Use MarkBind variables
+
+While authoring content, you can insert variables anywhere in your site. MarkBind treats any text surrounded with double curly braces as a variable, e.g. <code>{<span></span>{variable_name}}</code>. It then replaces variables with pre-defined values when generating the site.
```
Comment (acjh) [inline docs/userGuide/contentAuthoring.md]: Combine the first and last sentences into:

"You can insert variables to be replaced by pre-defined values anywhere in your site."

**Comment by Chng-Zhi-Xuan** [comment]
- Applied changes requested
- Refactored `_setTimestampVariable` method from Page.js to Site.js

File: docs/userGuide/contentAuthoring.md [hunk 2]
```diff
@@ -113,6 +113,34 @@ In addition, it supports:
...
+#### Glyphicons
+
+This asset is provided by [Glyphicons](https://glyphicons.com/) via Bootstrap.
+<br><br>
+<tip-box>
+Traditional usage:  
+`<span class='glyphicon glyphicon-hand-right' aria-hidden='true'></span>`
+<br>  
+MarkBind usage:  
+<code>{<span></span>{glyphicon-hand-right}}</code>
```
Comment (yamgent) [inline docs/userGuide/contentAuthoring.md]: Should be `glyphicon_hand_right` (underscroll)

File: docs/userGuide/contentAuthoring.md [hunk 3]
```diff
@@ -113,6 +113,34 @@ In addition, it supports:
...
+</tip-box>
+
+Use these [available glyphs](https://getbootstrap.com/docs/3.3/components/#glyphicons) on your site by surrounding a glyph's name with double curly braces.  
+e.g. <code>{<span></span>{glyphicon_hand_right}}</code> will be rendered as <code><span class='glyphicon glyphicon-hand-right' aria-hidden='true'></span></code>
+
+Note: Replace all instances of `-` with `_` in the glyph's name, e.g. <code>{<span></span>{glyphicon-hand-right}}</code> becomes <code>{<span></span>{glyphicon_hand_right}}</code>
```
Comment (yamgent) [inline docs/userGuide/contentAuthoring.md]: Full stop at the end.

Comment (acjh) [reply docs/userGuide/contentAuthoring.md]: The rest of the page doesn't use a full stop at the end.

**Comment by Chng-Zhi-Xuan** [comment]
- Apply changes requested.
- Add additional line for the timestamp section in UserGuide.

**Comment by acjh** [comment]
Not rendered on Netlify preview?
https://deploy-preview-222--markbind.netlify.com/userguide/contentauthoring#timestamp

Also, avoid commit messages like "Update UserGuide".
Suggestion: "Add usage of timestamp to User Guide"

**Comment by yamgent** [comment]
> Not rendered on Netlify preview?
> https://deploy-preview-222--markbind.netlify.com/userguide/contentauthoring#timestamp

I think that's because the Netlify's markbind doesn't support the timestamp syntax yet.

Should we hardcode the sample output instead?

**Comment by acjh** [comment]
> I think that's because the Netlify's markbind doesn't support the timestamp syntax yet.

Ah right, that has happened before: https://github.com/MarkBind/markbind/pull/196#issuecomment-383380418
@nicholaschuayunzhi Any idea if `npm link` would work in the build command?

> Should we hardcode the sample output instead?

As discussed last week, I think not.

**Comment by nicholaschuayunzhi** [comment]
Netlify build installs the latest version of markbind released on npm. I believe we discussed that the idea was to have the user documentation only on the latest version. When published, we rebuild user docs.

**Comment by nicholaschuayunzhi** [comment]
I think its possible to build with branch's version of MarkBind for pr only. Maybe thats preferred?

**Comment by damithc** [comment]
Preview using netlify: the **general setup** (i.e., how we instruct others to set up netlify previews ofr their own projects) should use the latest released version even for PR reviews. That is because an external user will only know of the latest published version.

Setting up netlify previews for the markbind repo itself _can_ (if possible) use unreleased features, especially for PR reviews.

File: docs/userGuide/contentAuthoring.md [hunk 4]
```diff
@@ -113,6 +113,35 @@ In addition, it supports:
 
 * More media blocks, embedding services and additional options can be found in [Markdown-it-block-embed docs](https://github.com/rotorz/markdown-it-block-embed)
 
+### Use MarkBind variables
+
+While authoring content, you can insert variables to be replaced by pre-defined values anywhere in your site. MarkBind treats any text surrounded with double curly braces as a variable, e.g. <code>{<span></span>{variable_name}}</code>
```
[the remark this answers is not in the document]
Comment (acjh) [reply docs/userGuide/contentAuthoring.md]: @Chng-Zhi-Xuan In case you missed this.

**Comment by nicholaschuayunzhi** [comment]
Did the styling of the drop down menu change?
Preview:
image

Live:
image

**Comment by acjh** [comment]
> Did the styling of the drop down menu change?

Yes. This branch is not on the tip of `master` but GitHub says it has no conflicts, so it should be fine.

**Comment by Chng-Zhi-Xuan** [comment]
- Squashed changes requested with older commits.

**Pull request merged**